protocol/0082-RFPR-on_chain_referral_program.md

```diff
@@ -0,0 +1,235 @@
+# On Chain Referral Program Specification
+
+## Summary
+
+The aim of the on-chain referral program is to incentivise users and community members to refer new traders by providing benefits for both referrers and referees. 
+
+Whilst a referral program is active, the following benefits will be available to eligible parties:
+
+- a **referrer** will receive a proportion of all referee taker fees as a **reward**.
+- a **referee** will be eligible for a **discount** on any taker fees they incur.
+
+The size of the reward or discount will be dependent on the benefit tier the referees [team](#team-mechanics) is currently placed in. A team becomes eligible for greater benefits by reaching a minimum running volume over a number of epochs. Both referrers and referees contribute to their teams running volume.
+
+On-chain referral programs can be created and updated through [governance proposals](#governance-proposals).
+
+## Network Parameters
+
+- `referralProgram.maxBenefitTiers` - limits the maximum number of [benefit tiers](#benefit-tiers) which can be specified as part of a referral program
+- `referralProgram.maxReferralRewardFactor` - limits the maximum reward factor which can be specified as part of a referral program
+- `referralProgram.maxReferralDiscountFactor` - limits the maximum discount factor which can be specified as part of a referral program governance proposal
+- `referralProgram.maxPartyVolumePerEpoch` - limits the volume which is eligible each epoch for referral program mechanisms
+- `referralProgram.minStakedVegaTokens` - limits team creation to parties staking at least this number of tokens
+
+## Governance Proposals
+
+An on-chain referral program can be proposed via governance. As part of the proposal, the proposer specifies the following fields:
+
+- `name`: name of the referral program
+- `benefit_tiers`: a list of dictionaries with the following fields
+  - `minimum_running_volume`: the required `running_team_volume` for a team to access this tier
+  - `referral_reward_factor`: the proportion of the referees taker fees to be rewarded to the referrer
+  - `referral_discount_factor`: the proportion of the referees taker fees to be discounted
+- `start_epoch`: the first epoch the program will become `STATE_ACTIVE` and benefits will be enabled
+- `end_epoch`: the first epoch the program will become `STATE_CLOSED` and benefits will be disabled
```

Comment:
Intuitively to me this would be the last active epoch rather than the first inactive